src/test/java/org/broadinstitute/consent/http/db/DacDAOTest.java

```diff
@@ -268,4 +272,49 @@ public void testFindDacsForDatasetIds() {
         assertFalse(dacs.isEmpty());
         assertEquals(1, dacs.size());
     }
+
+    @Test
+    public void testFindDatasetsAssociatedWithDac_AssignedDacId() {
+        Dataset datasetNonDac = createDataset();
+        Dac dac = createDac();
+
+        List<Dataset> results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(0, results.size());
```

Comment:
This block could be its own test to ensure that we're not finding unassigned datasets. Since this is also repeated in the next test, it makes even more sense to extract it. Each test should be focused on testing a single condition. Adding multiple conditions in a test case is generally an anti-pattern (not always, but usually).

Reply:
very good point - have updated all 3 tests to be more specific